dividat/driver
# Connect to Senso flex via bluetooth

Monitor bluetooth devices in a daemon, in order to keep a list of connected devices.

After having scanned serial ports, try to connect via Bluetooth to a Senso Flex device.

## Testing

1. Enable Bluetooth in OS, in nixos: `hardware.bluetooth.enable = true;`.
2. Activate bluetooth in network settings.
3. Build and run the driver.
4. Run Play at `dev-play.dividat.com`.
5. Power on Senso Flex without using the USB link, the led should blink.
6. After some seconds, the led should be fixed, this means Senso Flex is connected.

## Improvements

- Powering off Senso Flex, and then powering on again, it takes a long time to use it again, because the socket timeout is long.
- The flush function in some places, this works but it may not be the best solution.
- Logging in bluetooth monitoring may follow what’s used in other places in the app.

## Checklist

-   [x] Code documented
